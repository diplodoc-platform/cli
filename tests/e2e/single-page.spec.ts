import {describe, test} from 'vitest';
import {TestAdapter, compareDirectories, getTestPaths} from '../fixtures';

const generateMapTestSinglePageTemplate = (
    testTitle: string,
    testRootPath: string,
    {md2md = true, md2html = true, args = '--single-page'},
) => {
    test(testTitle, async () => {
        const {inputPath, outputPath} = getTestPaths(testRootPath);
        await TestAdapter.testBuildPass(inputPath, outputPath, {md2md, md2html, args});
        await compareDirectories(outputPath);
    });
};

describe('Single page mode', () => {
    generateMapTestSinglePageTemplate(
        'simple md2html single page with lang dirs',
        'mocks/single-page',
        {md2md: false},
    );
});
