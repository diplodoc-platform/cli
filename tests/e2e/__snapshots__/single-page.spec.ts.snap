// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Single page mode > simple md2html single page with lang dirs > filelist 1`] = `
"[
  "index.html",
  "ru/index.html",
  "ru/page.html",
  "ru/project/config.html",
  "ru/single-page-toc.js",
  "ru/single-page.html",
  "ru/single-page.json",
  "ru/toc.js"
]"
`;

exports[`Single page mode > simple md2html single page with lang dirs 1`] = `
"<!DOCTYPE html>
<html lang="ru" dir="ltr">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <base href="./" />
        <title>Redirect to ./ru/index.html</title>
        <meta http-equiv="refresh" content="0; url=./ru/index.html">
        <style type="text/css">html, body {min-height:100vh; height:100vh;}</style>
        <script type="application/javascript">
            window.location.replace("./ru/index.html");
        </script>
    </head>
    <body class="g-root g-root_theme_light">
        If you are not redirected automatically, follow this <a href="./ru/index.html">link</a>.
    </body>
</html>"
`;

exports[`Single page mode > simple md2html single page with lang dirs 2`] = `
"<!DOCTYPE html>
<html lang="ru" dir="ltr">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <base href="../" />
        <title>Documentation | Documentation</title>
        <link rel="canonical" href="ru/index.html">
        <link rel="alternate" href="ru/index.html" hreflang="ru" />
        <meta name="generator" content="Diplodoc Platform vDIPLODOC-VERSION">
        <style type="text/css">html, body {min-height:100vh; height:100vh;}</style>
    </head>
    <body class="g-root g-root_theme_light">
        <div id="root"></div>
        <script type="application/json" id="diplodoc-state">
            {"data":{"leading":true,"data":{"title":"Documentation","description":"","links":[{"title":"Getting started with Documentation","description":"This guide will show you the basics of working with Documentation","href":"ru/page.html"}]},"meta":{"metadata":[{"name":"generator","content":"Diplodoc Platform vDIPLODOC-VERSION"}],"alternate":[{"href":"ru/index.html","hreflang":"ru"}],"canonical":"ru/index.html","title":"Documentation","noIndex":true,"vcsPath":"ru/index.yaml"},"title":"Documentation"},"router":{"pathname":"ru/index","depth":2,"base":"../"},"lang":"ru","langs":["ru"],"viewerInterface":{"toc":true,"search":true,"feedback":true}}
        </script>
        <script type="application/javascript">
            const data = document.querySelector('script#diplodoc-state');
            window.__DATA__ = JSON.parse((function unescape(string) {
                return string.replace(/&lt;/g, "<").replace(/&gt;/g, ">").replace(/&amp;/g, "&");
              })(data.innerText));
            window.STATIC_CONTENT = false;
        </script>
        <script type="application/javascript" defer src="ru/toc.js"></script>
    </body>
</html>"
`;

exports[`Single page mode > simple md2html single page with lang dirs 3`] = `
"<!DOCTYPE html>
<html lang="ru" dir="ltr">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <base href="../" />
        <title>Page Title | Documentation</title>
        <link rel="canonical" href="ru/page.html">
        <link rel="alternate" href="ru/page.html" hreflang="ru" />
        <meta name="generator" content="Diplodoc Platform vDIPLODOC-VERSION">
        <meta name="yfm" content="builder in page">
        <meta name="description" content="Some test description">
        <style type="text/css">html, body {min-height:100vh; height:100vh;}</style>
        <link rel="icon" type="image/x-icon" href="/favicon.ico">
    </head>
    <body class="g-root g-root_theme_light">
        <div id="root"></div>
        <script type="application/json" id="diplodoc-state">
            {"data":{"leading":false,"html":"&lt;p&gt;Lorem&lt;/p&gt;/n","meta":{"metadata":[{"name":"generator","content":"Diplodoc Platform vDIPLODOC-VERSION"},{"name":"yfm","content":"builder in page"},{"name":"description","content":"Some test description"}],"alternate":[{"href":"ru/page.html","hreflang":"ru"}],"canonical":"ru/page.html","title":"Page Title","description":"Some test description","interface":{"toc":false,"favicon-src":"/favicon.ico"},"vcsPath":"ru/page.md"},"headings":[],"title":"Page Title"},"router":{"pathname":"ru/page","depth":2,"base":"../"},"lang":"ru","langs":["ru"],"viewerInterface":{"toc":false,"search":true,"feedback":true,"favicon-src":"/favicon.ico"}}
        </script>
        <script type="application/javascript">
            const data = document.querySelector('script#diplodoc-state');
            window.__DATA__ = JSON.parse((function unescape(string) {
                return string.replace(/&lt;/g, "<").replace(/&gt;/g, ">").replace(/&amp;/g, "&");
              })(data.innerText));
            window.STATIC_CONTENT = false;
        </script>
        <script type="application/javascript" defer src="ru/toc.js"></script>
    </body>
</html>"
`;

exports[`Single page mode > simple md2html single page with lang dirs 4`] = `
"<!DOCTYPE html>
<html lang="ru" dir="ltr">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <base href="../../" />
        <title>Documentation</title>
        <link rel="canonical" href="ru/project/config.html">
        <link rel="alternate" href="ru/project/config.html" hreflang="ru" />
        <meta name="generator" content="Diplodoc Platform vDIPLODOC-VERSION">
        <style type="text/css">html, body {min-height:100vh; height:100vh;}</style>
    </head>
    <body class="g-root g-root_theme_light">
        <div id="root"></div>
        <script type="application/json" id="diplodoc-state">
            {"data":{"leading":false,"html":"&lt;p&gt;Lorem&lt;/p&gt;/n","meta":{"metadata":[{"name":"generator","content":"Diplodoc Platform vDIPLODOC-VERSION"}],"alternate":[{"href":"ru/project/config.html","hreflang":"ru"}],"canonical":"ru/project/config.html","vcsPath":"ru/project/config.md"},"headings":[],"title":""},"router":{"pathname":"ru/project/config","depth":3,"base":"../../"},"lang":"ru","langs":["ru"],"viewerInterface":{"toc":true,"search":true,"feedback":true}}
        </script>
        <script type="application/javascript">
            const data = document.querySelector('script#diplodoc-state');
            window.__DATA__ = JSON.parse((function unescape(string) {
                return string.replace(/&lt;/g, "<").replace(/&gt;/g, ">").replace(/&amp;/g, "&");
              })(data.innerText));
            window.STATIC_CONTENT = false;
        </script>
        <script type="application/javascript" defer src="ru/toc.js"></script>
    </body>
</html>"
`;

exports[`Single page mode > simple md2html single page with lang dirs 5`] = `"window.__DATA__.data.toc = {"title":"Documentation","href":"ru/single-page.html#ru_index","items":[{"name":"Documentation","href":"ru/single-page.html#ru_page","id":"UUID"},{"name":"Config","href":"ru/single-page.html#ru_project_config","id":"UUID"}],"path":"ru/toc.yaml","id":"UUID"};"`;

exports[`Single page mode > simple md2html single page with lang dirs 6`] = `
"<!DOCTYPE html>
<html lang="ru" dir="ltr">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <base href="../" />
        <title>Documentation | Documentation</title>
        <meta name="generator" content="Diplodoc Platform vDIPLODOC-VERSION">
        <meta name="yfm" content="builder in page">
        <meta name="description" content="Some test description">
        <style type="text/css">html, body {min-height:100vh; height:100vh;}</style>
    </head>
    <body class="g-root g-root_theme_light">
        <div id="root"></div>
        <script type="application/json" id="diplodoc-state">
            {"data":{"leading":false,"html":"&lt;h2 data-original-article=\\"ru/page.html\\"&gt;Page Title&lt;a class=\\"yfm-anchor\\" aria-hidden=\\"true\\" href=\\"ru/single-page.html#ru_page\\" id=\\"ru_page\\"&gt;&lt;/a&gt;&lt;/h2&gt;&lt;p&gt;Lorem&lt;/p&gt;/n&lt;hr class=\\"yfm-page__delimeter\\"&gt;&lt;p&gt;Lorem&lt;/p&gt;/n","headings":[],"meta":{"metadata":[{"name":"generator","content":"Diplodoc Platform vDIPLODOC-VERSION"},{"name":"yfm","content":"builder in page"},{"name":"description","content":"Some test description"}]},"title":"Documentation"},"router":{"pathname":"ru/single-page","depth":2,"base":"../"},"lang":"ru","langs":["ru"],"viewerInterface":{"toc":true,"search":true,"feedback":true}}
        </script>
        <script type="application/javascript">
            const data = document.querySelector('script#diplodoc-state');
            window.__DATA__ = JSON.parse((function unescape(string) {
                return string.replace(/&lt;/g, "<").replace(/&gt;/g, ">").replace(/&amp;/g, "&");
              })(data.innerText));
            window.STATIC_CONTENT = false;
        </script>
        <script type="application/javascript" defer src="ru/single-page-toc.js"></script>
    </body>
</html>"
`;

exports[`Single page mode > simple md2html single page with lang dirs 7`] = `"{"data":{"leading":false,"html":"<h2 data-original-article=\\"ru/page.html\\">Page Title<a class=\\"yfm-anchor\\" aria-hidden=\\"true\\" href=\\"ru/single-page.html#ru_page\\" id=\\"ru_page\\"></a></h2><p>Lorem</p>/n<hr class=\\"yfm-page__delimeter\\"><p>Lorem</p>/n","headings":[],"meta":{"metadata":[{"name":"generator","content":"Diplodoc Platform vDIPLODOC-VERSION"},{"name":"yfm","content":"builder in page"},{"name":"description","content":"Some test description"}]},"title":"Documentation","toc":{"title":"Documentation","href":"ru/index.html","items":[{"name":"Documentation","href":"ru/page.html","id":"UUID"},{"name":"Config","href":"ru/project/config.html","id":"UUID"}],"path":"ru/toc.yaml","id":"UUID"}},"router":{"pathname":"ru/single-page","depth":2,"base":"../"},"lang":"ru","langs":["ru"],"viewerInterface":{"toc":true,"search":true,"feedback":true}}"`;

exports[`Single page mode > simple md2html single page with lang dirs 8`] = `"window.__DATA__.data.toc = {"title":"Documentation","href":"ru/index.html","items":[{"name":"Documentation","href":"ru/page.html","id":"UUID"},{"name":"Config","href":"ru/project/config.html","id":"UUID"}],"path":"ru/toc.yaml","id":"UUID"};"`;
